src/main/java/classes/Duke.java

```diff
@@ -0,0 +1,77 @@
+package classes;
+
+import exceptions.IncorrectNoOfArgumentException;
+
+import java.io.IOException;
+
+import java.util.ArrayList;
+
+import java.time.format.DateTimeParseException;
+
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    public Duke(String filePath, String folderPath) {
+        this.ui = new Ui();
+        this.storage = new Storage(filePath, folderPath);
+        try {
+            this.tasks = new TaskList(storage.load());
+        } catch (IOException e) {
+            this.ui.showLoadingError(e);
+            this.tasks = new TaskList(new ArrayList<String>());
+        }
+    }
+
+    public void run() throws IOException {
+        ui.showWelcome();
+        boolean isExit = false;
+        while (!isExit) {
+            ArrayList<String> commandInfo;
+            try {
+                commandInfo = Parser.parse(ui.readCommand());
+            } catch (IncorrectNoOfArgumentException e) {
+                System.out.println(e);
+                continue;
+            }
+            if (commandInfo.size() != 0) {
+                switch (commandInfo.get(0)) {
```

Comment:
Switch statement is well-formatted according to the coding standard (i.e. no indentation for case clauses). Good job! 😄